compiler/src/typed/subst.re

```diff
@@ -132,16 +150,23 @@ let norm =
 
 let ctype_apply_env_empty = ref(_ => assert(false));
 
+let ident = (s, x) => Ident.rename(x);
+let rec path = (s, x) =>
+  switch (x) {
+  | PIdent(id) => PIdent(ident(s, id))
+  | PExternal(mod_, name) => PExternal(path(s, mod_), name)
+  };
+
 /* Similar to [Ctype.nondep_type_rec]. */
 let rec typexp = (s, ty) => {
   let ty = repr(ty);
   switch (ty.desc) {
   | (TTyVar(_) | TTyUniVar(_)) as desc =>
     /* Is this okay? This causes serialized type variables to lose their proper IDs. -Oscar */
-    /* if s.for_saving || ty.id < 0 then */
-    if (s.for_saving) {
+    /* if s.for_cmi || ty.id < 0 then */
```

Comment:
Still haven't answered this question? 😛

Reply:
No 😛

Theoretically if we ever figured this out we wouldn't need the `Translsig` module. I think? The world may never know 🙈